articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,185 @@
+    training_ratio = 0.75
+    training_data= []
+    evaluation_data = []
+
+    for indice in range(0,total):
+        if indice<total*training_ratio:
+            training_data.append(data[indice])
+        else:
+            evaluation_data.append(data[indice])
+
+    return training_data, evaluation_data
+```
+
+Now we are going to create two functions, one will be *preprocessing_step* that will get the data, preprocess it, and then split it, and the second one will be *training_step* that will receive two parameters and vectorize the training text.
+
+```python
+@st.cache
+def preprocessing_step():
+    data = get_all_data()
+    processing_data = preprocessing_data(data)
+    return split_data(processing_data)
+
+def training_step(data,vectorizer):
+    training_text = [data[0] for data in data]
+    training_result = [data[1] for data in data]
+    training_text = vectorizer.fit_transform(training_text)
+
+    return BernoulliNB().fit(training_text,training_result)
+```
+The next step will be passing the training and evaluation data into the *preprocessing_step* function, choosing a vectorizer( In this case we will be using CountVecotizer), and finally, a classifier variable that will call the *training_step* function and pass in the training_data and vectorizer.
+
+```python
+training_data,evaluation_data = preprocessing_step()
+vectorizer = CountVectorizer(binary='true')
+classifier = training_step(training_data,vectorizer)
+```
+After carrying out the above step the next step will be writing two functions, the first will be *analyze_text* and it will carry out the analysis by taking in the classifier, vectorizer, and text value, and the second function will be called *print_result* and will send a result as positive or negative.
+
+```python
+def analyse_text(classifier,vectorizer,text):
+    return text,classifier.predict(vectorizer.transform([text]))
+
+def print_result(result):
+    text,analysis_result = result
+    print_text = "Positive" if analysis_result[0]=='1' else "Negative"
+    return text,print_text
+```
+Now it’s time for us to continue building the interface, we will simply need an input form, button, and output textbox.
+
+```python
+review = st.text_input("Enter The Review","Write Here...")
+if st.button('Predict Sentiment'):
+    result = print_result(analyse_text(classifier,vectorizer,review))
+    st.success(result[1])
+else:
+    st.write("Press the above button..")
+```
+The codes above use streamlit *st.text_input* to create a text input form that will display “enter the review, write here”, the if *st.button* will create a button that will display “predict sentiment”, after that the variable result calls the *print_result* function and pass in the parameters. If the whole process the st.sucess(result[1) displays the result else it passes an error message asking the user to press the above button. Run your app using the command below.
+
+```python
+streamit run textanalyzer.py
+```
+#### Conclusion
```

Comment:
Include the final output jpeg/png image. How will it look? Take a screenshot of your terminal and the web app.